src/main/kotlin/net/pwall/json/schema/parser/Parser.kt

```diff
@@ -129,7 +129,12 @@ class Parser(var options: Options = Options(), uriResolver: (URI) -> InputStream
         return parse(json, uri)
     }
 
-    private fun parse(json: JSONValue, uri: URI): JSONSchema {
+    fun parse(string: String): JSONSchema {
+        val json = jsonReader.readJSON(string)
+        return parse(json,null)
```

Comment:
I might suggest one very small change simply to match the function invocations in the rest of the file. Thanks for putting this up! It was very helpful to see and I hope it can be merged. 
```suggestion
        return parse(json, null)
```

Reply:
Thanks for the suggestion, I have edited it.